src/microbe_stage/CellHexesPhotoBuilder.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using Godot;
+
+public class CellHexesPhotoBuilder : Spatial, IPhotographable
+{
+    private float radius;
+    private bool radiusDirty;
+    private MicrobeSpecies? species;
+
+    public string SceneToPhotographPath => "res://src/microbe_stage/CellHexesPhotoBuilder.tscn";
+
+    public float Radius
+    {
+        get
+        {
+            if (radiusDirty)
+                CalculateRadius();
+
+            return radius;
+        }
+    }
+
+    public MicrobeSpecies? Species
+    {
+        get => species;
+        set
+        {
+            species = value;
+            radiusDirty = true;
+        }
+    }
+
+    public void ApplySceneParameters(Spatial instancedScene)
+    {
+        var builder = (CellHexesPhotoBuilder)instancedScene;
+        builder.Species = Species;
+        builder.BuildHexStruct();
+    }
+
+    public float CalculatePhotographDistance(Spatial instancedScene)
+    {
+        return PhotoStudio.CameraDistanceFromRadiusOfObject(((CellHexesPhotoBuilder)instancedScene).Radius *
+            Constants.PHOTO_STUDIO_CELL_RADIUS_MULTIPLIER);
+    }
+
+    private void BuildHexStruct()
+    {
+        var hexScene = GD.Load<PackedScene>("res://src/microbe_stage/editor/EditorHex.tscn");
+        var hexMaterial = GD.Load<Material>("res://src/microbe_stage/editor/ValidHex.material");
+        var modelScene = GD.Load<PackedScene>("res://src/general/SceneDisplayer.tscn");
+
+        this.QueueFreeChildren();
+
+        if (Species == null)
+            throw new InvalidOperationException("Species is ont initialized");
```

Comment:
```suggestion
            throw new InvalidOperationException("Species is not initialized");
```
one more typo I spotted